addons/base_import/models/base_import.py

```diff
@@ -626,6 +637,21 @@ def _parse_import_data(self, data, import_fields, options):
                 # We should be able to manage both case
                 index = import_fields.index(name)
                 self._parse_float_from_data(data, index, name, options)
+            elif field['type'] == 'binary' and name in import_fields:
+                index = import_fields.index(name)
+                for line in data:
+                    url_regex = re.compile(
+                        r'^ftp|http|https|smb?://'  # http:// or https:// or ftp:// or smb://
```

Comment:
1. why would we support ftp and smb?
2. why compile the regex only to immediately match on it? Just use re.match(pattern, string, flags)
3. the regex is broken, it will match `ftp` (just ftp without anything else) or `sm://`
4. at this point you might as well just check if the value starts with the relevant schemes